shoes-core/lib/shoes/common/visibility.rb

```diff
@@ -19,21 +19,21 @@ def visible?
 
       # Reveals the element, if it is hidden. See also #hide and #toggle.
       def show
-        @hidden = false
+        style[:hidden] = false
         update_visibility
       end
 
       # Hides an element if it is shown. Or shows the element, if it is hidden.
       # See also #hide and #show.
       def toggle
-        @hidden = !@hidden
+        style[:hidden] = !style[:hidden]
         update_visibility
       end
 
       private
 
       def update_visibility
-        gui.update_visibility
+        gui.update_visibility if gui.respond_to?(:update_visibility)
```

Comment:
I'd be against having this check in here just for the mock backend. There should be a `CommonMethods` somewhere in the mock world, I'd just jam the `update_visibility` method in there.